src/client/eppo-client.ts

```diff
@@ -82,168 +82,174 @@ export default class EppoClient implements IEppoClient {
   private queuedEvents: IAssignmentEvent[] = [];
   private assignmentLogger: IAssignmentLogger | undefined;
 
-  constructor(private configurationStore: IConfigurationStore) {}
+  constructor(private configurationStore: IConfigurationStore) { }
 
   public getAssignment(
     subjectKey: string,
-    experimentKey: string,
+    flagKey: string,
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
     subjectAttributes: Record<string, any> = {},
     assignmentHooks?: IAssignmentHooks | undefined,
   ): string | null {
-    const assignment = this.getAssignmentInternal(
+    const { allocationKey, assignment } = this.getAssignmentInternal(
       subjectKey,
-      experimentKey,
+      flagKey,
       subjectAttributes,
       assignmentHooks,
       ValueType.StringType,
     );
-    assignmentHooks?.onPostAssignment(experimentKey, subjectKey, assignment);
+    assignmentHooks?.onPostAssignment(flagKey, subjectKey, assignment, allocationKey);
 
-    if (assignment !== null)
-      this.logAssignment(experimentKey, assignment, subjectKey, subjectAttributes);
+    if (assignment !== null && allocationKey !== null)
```

Comment:
`allocationKey` will be `null` for overrides returned by `getAssignmentInternal`, so this is how prevent logging an assignment for an overridden subject. Nice. Howeve,r I think before this change, this SDK was logging overrides 😬

Reply:
Hmmm I'm thinking about this more and now wondering if I should keep the original functionality of logging overrides ..

So there's the `onPreAssignment` function, which is one way to override the assignment

And there's subject overrides (see `getSubjectVariationOverride`) which uses the "overrides" attribute in the RAC, which seems to come from `opted_in_subjects` in the `experiment_variations` table, which I checked and the only company using this is Goldbelly for one specific variation. Also I don't see any UI for supporting this, so this seems to be a pretty custom thing that we built for this one specific purpose.

The `onPreAssignment` call doesn't exist in any of the other SDKs and again there's one variation by one company that's actually using the overrides, so maybe this is all an edge case that doesn't require us to actively work on making all the other SDKs consistent, as far as whether we log or don't log overrides.

BUT I feel like any customer using these features might actually want to see these overridden assignments logged (or how else would they know that the overrides are working ..). So I'm going to change this back to let `null` allocation keys get logged

Reply:
Actually nevermind, I don't know what to do with the experiment key when there's no allocation key. I can't leave out the allocation key because we're already doing a `'${flagKey}' or '${flagKey}-${allocationKey}'` to find experiment assignments and putting in any placeholder text for the allocationKey like '${flagKey}-override` is fragile if anyone makes a feature flag key that ends with '-override'

Maybe we can take the stance that overrides are not assignments and so we leave them out of analysis and we leave it to the customers to track this information if they want it, e.g. with the onPostAssignment hook